pandas/core/indexes/multi.py

```diff
@@ -2165,7 +2165,11 @@ def append(self, other):
             arrays = []
             for i in range(self.nlevels):
                 label = self._get_level_values(i)
-                appended = [o._get_level_values(i) for o in other]
+                if label.names[0] and concat_indexes == True:
```

Comment:
doesn't `label=self._get_level_values(i)` return a single index? In which case it may or may not have a `name` so maybe `.names[0]` is unnecessary, can just use `.name`